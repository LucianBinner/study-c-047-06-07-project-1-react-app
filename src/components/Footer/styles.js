import styled from 'styled-components';
import { theme } from '../../styles/theme';

export const Container = styled.div`
  text-align: center;

  a {
    color: inherit;
    text-decoration: none;
    font-size: ${theme.font.sizes.small};
  }
`;
