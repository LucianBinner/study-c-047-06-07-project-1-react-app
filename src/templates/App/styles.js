import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme, background }) => css`
    background: ${background};
    color: ${theme.colors.mainBg};
  `}
`;
