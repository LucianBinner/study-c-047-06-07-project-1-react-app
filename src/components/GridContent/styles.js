import styled from 'styled-components';
import { theme } from '../../styles/theme';

export const Container = styled.div`
  text-align: center;
  max-width: 55rem;
  margin: 0 auto;
`;

export const Html = styled.div`
  margin: ${theme.spacings.xhuge} 0;
`;
