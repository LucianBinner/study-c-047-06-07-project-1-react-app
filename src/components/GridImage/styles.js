import styled from 'styled-components';
import { theme } from '../../styles/theme';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  > ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
    }
`;

export const Grid = styled.div`
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  gap: ${theme.spacings.large};
`;

export const GridElement = styled.div`
  overflow: hidden;
`;

export const Image = styled.img`
  width: 100%;
  transition: all 300ms ease-in-out;

  &:hover {
    transform: scale(1.2) rotate(10deg);
  }
`;
