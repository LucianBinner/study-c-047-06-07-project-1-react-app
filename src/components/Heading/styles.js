import styled, { css } from 'styled-components';
import { theme } from '../../styles/theme';

const titleSize = {
  small: () => css`
    font-size: ${theme.font.sizes.medium};
  `,
  medium: () => css`
    font-size: ${theme.font.sizes.large};
  `,
  big: () => css`
    font-size: ${theme.font.sizes.xlarge};
  `,
  huge: () => css`
    font-size: ${theme.font.sizes.xhuge};
    ${mediaFont(theme)};
  `,
};

const mediaFont = () => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.font.sizes.xlarge};
  }
`;

const titleCase = (uppercase) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1`
  color: ${(props) =>
    props.colorDark ? theme.colors.primaryColor : theme.colors.white};

  ${({ size }) => titleSize[size]()}
  ${({ uppercase }) => titleCase(uppercase)}
`;
