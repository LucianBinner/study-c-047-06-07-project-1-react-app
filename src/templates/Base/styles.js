import styled from 'styled-components';
import { theme } from '../../styles/theme';

export const Container = styled.div`
  padding-top: 5.4rem;
  width: 100%;

  @media ${theme.media.lteMedium} {
    padding-top: 0;
  }
`;
