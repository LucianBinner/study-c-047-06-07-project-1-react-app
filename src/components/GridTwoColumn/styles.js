import styled from 'styled-components';
import { theme } from '../../styles/theme';
import { Title } from '../Heading/styles';

export const Container = styled.div`
  display: grid;
  grid-template-columns: 1fr 2fr;
  align-items: center;
  gap: ${theme.spacings.large};

  @media ${theme.media.lteMedium} {
    grid-template-columns: 1fr;
    text-align: center;
  }

  ${Title} {
    margin-bottom: ${theme.spacings.large};
  }
`;

export const TextContainer = styled.div`
  @media ${theme.media.lteMedium} {
    margin-bottom: ${theme.spacings.large};
  }
`;

export const ImageContainer = styled.div``;

export const Image = styled.img``;
