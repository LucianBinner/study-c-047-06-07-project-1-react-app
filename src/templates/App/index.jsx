import { Heading } from '../../components/Heading';
import { Wrapper } from './styles';

const App = () => {
  return (
    <>
      <Heading>Header</Heading>
      <Wrapper background="blue">
        <h1>Hello World!</h1>
      </Wrapper>
    </>
  );
};

export default App;
