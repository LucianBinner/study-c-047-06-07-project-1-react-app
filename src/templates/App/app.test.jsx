import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import App from '.';

test('Renders learn react link', () => {
  renderTheme(<App />);
  const headingContainer = screen.getByRole('heading', {
    name: 'Hello World!',
  }).parentElement;
  expect(headingContainer).toHaveStyle({
    background: 'blue',
  });
  expect(headingContainer).toMatchSnapshot(); // Faz snapchot do elemento testado
  expect(headingContainer).toHaveStyleRule('background', 'blue');
});
