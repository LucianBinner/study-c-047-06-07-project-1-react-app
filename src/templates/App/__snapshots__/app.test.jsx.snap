// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders learn react link 1`] = `
.c0 {
  background: blue;
  color: green;
}

<div
  class="c0"
>
  <h1>
    Hello World!
  </h1>
</div>
`;
